lib/utility.rb

```diff
@@ -135,7 +135,9 @@ def prompt(message)
 # thanks to https://github.com/devopsgroup-io/vagrant-hostmanager/issues/121#issuecomment-69050265
 
 def read_ip_address(machine)
-  command = "LANG=en ifconfig  | grep 'inet addr:'| grep -v '127.0.0.1' | cut -d: -f2 | awk '{ print $1 }'"
+  # workaround for public ip address: get the pattern 192.168.x.x
+  # this could be problems with IP address other than this pattern for LAN access
+  command = "LANG=en ifconfig | grep 'inet addr:192.168' | cut -d: -f2 | awk '{ print $1 }'"
```

Comment:
10.x.x.x is not accessible within the LAN network, is that correct?

Reply:
This is the workaround, because I could see eth0 and eth1 was not fixed, sometimes eth0 holds 10.x.x.x, sometimes it holds 192.168.x.x, so I introduces the workaround to bypass this problem.

I always expect 192.168.x.x for LAN accessible IP for the VM. Maybe we could have a better solution than this workaround?

Reply:
I mean sometime public interface could get 10.x.x.x, for example, LAN address is from 10.0.0.1~10.0.0.254 ( aka 10.0.0.0/24), while internal interface use 10.0.2.15~10.0.2.254 ( aka 10.0.2.0/24). 
With that case, the command will fail to `grep 'inet addr:192.168'`.

Reply:
I see, can we find a way to identify the LAN interface?

Reply:
The truth is I don't know. I tried to detect base on interface created order ( like you said about eth0, eth1, ensp0, ensp1....), base on IP, base on the configuration file of that interface created time ..... but the result is inconsistent and every box version has different network behavior. Only 2 things thinking about it now is (sorry it just a draft idea) :
+ Save the host's IP address info to VM. Detect VM public interface by compare it with host's network ( VM public interface will always have same network with host)
+ Other things is detecting base on network hardware id `lshw -c network -businfo | grep pci | grep network` my prediction is the higher number in pci@xxxx will be the public interface.

Reply:
The first idea is:
+ know host's bridge interface is X, found that X's ip address is belong to 10.0.1.0/24 network.
+ Compare 10.0.1.0/24 to IP of all local network of VM and found the interface Y that held the same network as 10.0.1.0/24. Y is the public interface. Print that IP address to console.